src/main/java/bio/terra/service/dataset/flight/delete/DeleteDatasetAzurePrimaryDataStep.java

```diff
@@ -0,0 +1,75 @@
+package bio.terra.service.dataset.flight.delete;
+
+import bio.terra.service.configuration.ConfigEnum;
+import bio.terra.service.configuration.ConfigurationService;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.filedata.azure.blobstore.AzureBlobStorePdao;
+import bio.terra.service.filedata.google.firestore.FireStoreDao;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+public class DeleteDatasetAzurePrimaryDataStep implements Step {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(DeleteDatasetAzurePrimaryDataStep.class);
+
+  private final AzureBlobStorePdao azureBlobStorePdao;
+  private final FireStoreDao fileDao;
+  private final DatasetService datasetService;
+  private final UUID datasetId;
+  private final ConfigurationService configService;
+
+  public DeleteDatasetAzurePrimaryDataStep(
+      AzureBlobStorePdao azureBlobStorePdao,
+      FireStoreDao fileDao,
+      DatasetService datasetService,
+      UUID datasetId,
+      ConfigurationService configService) {
+    this.azureBlobStorePdao = azureBlobStorePdao;
+    this.fileDao = fileDao;
+    this.datasetService = datasetService;
+    this.datasetId = datasetId;
+    this.configService = configService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException {
+    Dataset dataset = datasetService.retrieve(datasetId);
+    if (configService.testInsertFault(ConfigEnum.LOAD_SKIP_FILE_LOAD)) {
+      // If we didn't load files, don't try to delete them
+      fileDao.deleteFilesFromDataset(dataset, fireStoreFile -> {});
+    } else {
+      fileDao.deleteFilesFromDataset(dataset, azureBlobStorePdao::deleteFile);
```

Comment:
It looks like everything in this method is duplicated in `DeleteDatasetPrimaryDataStep`, except for this line. This is in partly due to the wordy fault insert code, which could be made DRY by moving it into `FileDao`.

Even if the fault code was moved, using a class hierarchy is one way to share the common code between two different steps, if we are totally against making this step internally conditional.

Reply:
This is going to look a lot different moving forward when we start having metadata not in BQ and Firestore